### ISSUE TITLE
Replacing goog.isObject

### ISSUE THREAD

**Issue by probins** [openlayers/ol3#5542]
does:
```
var type = typeof val;
return type == 'object' && val != null || type == 'function';
```
Reproduce this (or something similar) as `ol.isObject`?

**Comment by jonataswalker** [comment]
Maybe you want to exclude arrays:

```javascript
return (typeof val === 'object' || typeof val === 'function') && !Array.isArray(val) && val !== null;
```

**Comment by probins** [comment]
this is mainly used in asserts, otherwise only used in 3 places. Create `ol.isObject` or just reproduce inline?

**Comment by ahocevar** [comment]
I think the asserts are just there to satisfy the compiler. In the remaining cases, I'm wondering if it wouldn't be enough to just check for `type == 'object'`. @probins, were you planning to create a pull request?

**Comment by probins** [comment]
yes. If it's just checking for `type == 'object'`, that's very straightforward. There is though the bigger question of what is planned for `goog.asserts`.

**Comment by ahocevar** [comment]
For `goog.asserts`, any solution wrapped in an `ol.asserts` will do (e.g. use `console.assert`, throw an error, etc.). If we remove `goog.asserts`, many assertions will have to be replaced with type casts, because we only have them in the code to satisfy the compiler. All other assertions will work fine with any replacement.

The important thing is that assertions need to be stripped out in a build step, because we do not want to execute assertions in compiled code. Is this something that tools like browserify or whatever we might be using in the future can do?

**Comment by probins** [comment]
Don't think so. These tools will remove unused code - if you only use one import from a module that has five exports, Rollup for example will leave out the others - but in this case the code isn't unused. ISTM it would be better to replace assertions with type casts where possible, as type casts are just comments which any minifier will remove.

For the moment, I'll put together a PR to replace `goog.isObject`. We can deal with asserts separately.

**Comment by ahocevar** [comment]
Thinking through this again, I got reminded but `typeof null == 'object'` is true. So it is probably really better to introduce an `ol.isObject` function in `src/ol/objectutil.js`. `goog.isObject` also gives a positive for `typeof obj == 'function'`, but I don't think we need this.

**Comment by ahocevar** [comment]
> For the moment, I'll put together a PR to replace goog.isObject. We can deal with asserts separately.

Sounds good @probins. I hope my https://github.com/openlayers/ol3/issues/5542#issuecomment-232892097 above helps.

**Comment by probins** [comment]
OTOH if the asserts are just there to cast to an Object, then it would be simpler to deal with this in the PR replacing `goog.assert`. So maybe it would be better for me to just replace the 3 non-assert cases, and leave those in asserts?

**Comment by ahocevar** [comment]
@probins 

> So maybe it would be better for me to just replace the 3 non-assert cases, and leave those in asserts?

Yes please, that's a good idea. It will also make merging easier when I work on the goog.asserts removal in parallel.

**Comment by probins** [comment]
fixed by #5605; those in asserts can be handled separately
